modules/common/src/main/java/unnamed/mmo/lang/LanguageProvider.java

```diff
@@ -0,0 +1,47 @@
+package unnamed.mmo.lang;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextReplacementConfig;
+import org.jetbrains.annotations.NotNull;
+import unnamed.mmo.util.ComponentUtil;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+/**
+ * Naive component translation system.
+ * <p>
+ * Should be replaced with adventure translation or something else in the future.
+ */
+public class LanguageProvider {
+    private static final Properties properties = new Properties();
+
+    static {
+        try {
+            properties.load(LanguageProvider.class.getResourceAsStream("/lang/en_US.properties"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static final Pattern ARG_PATTERN = Pattern.compile("\\{[0-9]+}");
+
+    public static @NotNull Component get(@NotNull String key, @NotNull Component... args) {
```

Comment:
Can you add some JavaDocs and comments to this method? It's a bit unclear as to what it is meant to do